metricshub-hardware/src/main/java/org/sentrysoftware/metricshub/hardware/HardwareEnergyPostExecutionService.java

```diff
@@ -339,12 +347,72 @@ private void collectNetworkMonitorMetrics(final Monitor monitor) {
 			}
 		}
 
-		PowerAndEnergyCollectHelper.collectPowerAndEnergy(
-			monitor,
-			HW_POWER_NETWORK_METRIC,
-			HW_ENERGY_NETWORK_METRIC,
-			telemetryManager,
-			new NetworkPowerAndEnergyEstimator(monitor, telemetryManager)
-		);
+		if (connectorHasHardwareTag(monitor, telemetryManager)) {
+			PowerAndEnergyCollectHelper.collectPowerAndEnergy(
+				monitor,
+				HW_POWER_NETWORK_METRIC,
+				HW_ENERGY_NETWORK_METRIC,
+				telemetryManager,
+				new NetworkPowerAndEnergyEstimator(monitor, telemetryManager)
+			);
+		}
+	}
+
+	/**
+	 * Checks if the connector associated with the provided monitor has a "hardware" tag.
+	 *
+	 * @param monitor the monitor containing the connector ID attribute
+	 * @param telemetryManager the telemetry manager containing the connector store
+	 * @return true if the connector has a "hardware" tag, false otherwise
+	 */
+	private boolean connectorHasHardwareTag(final Monitor monitor, final TelemetryManager telemetryManager) {
+		if (monitor == null) {
+			return false;
+		}
+		final ConnectorStore telemetryManagerConnectorStore = telemetryManager.getConnectorStore();
+		if (telemetryManagerConnectorStore == null) {
+			log.error(
+				"Hardware Energy and Sustainability Module: Host {} connectorStore does not exist.",
+				telemetryManager.getHostname()
+			);
+			return false;
+		}
+
+		final Map<String, Connector> store = telemetryManagerConnectorStore.getStore();
+
+		if (store == null) {
+			log.error(
+				"Hardware Energy and Sustainability Module: Host {} connectorStore store does not exist.",
+				telemetryManager.getHostname()
+			);
+			return false;
+		}
+
+		final String connectorId = monitor.getAttribute(MONITOR_ATTRIBUTE_CONNECTOR_ID);
+
+		if (connectorId == null) {
+			log.error(
+				"Hardware Energy and Sustainability Module: Host {}. Monitor {} connector_id attribute does not exist.",
+				telemetryManager.getHostname(),
+				monitor.getId()
+			);
+			return false;
+		}
+
+		final Connector connector = store.get(connectorId);
+
+		if (connector == null) {
+			log.error(
+				"Hardware Energy and Sustainability Module: Host {}. Monitor {} connector_id attribute does not correspond to any valid connector id.",

```

Comment:
Update the message please to use the same pattern as the other log messages in MetricsHub.